sgt-back-end/index.js

```diff
@@ -0,0 +1,161 @@
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select "gradeId",
+         "name",
+         "course",
+         "score",
+         "createdAt"
+    from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An expected error occurred' });
+    });
+});
+
+app.post('/api/grades/', (req, res) => {
+  const postGrade = req.body;
+  const score = Number(postGrade.score);
+  if (!postGrade.name) {
+    res.status(400).json({ error: 'Must include a valid name' });
+  } else if (!postGrade.course) {
+    res.status(400).json({ error: 'Must include a valid course' });
+  } else if (!score) {
+    res.status(400).json({ error: 'Must include a valid score' });
+  } else if (score < 0 || score > 100 || isNaN(score)) {
+    res.status(400).json({ error: 'Score must be a positve integer between 0 & 100' });
```

Comment:
While this catches non-numbers and numbers out of range, it doesn't catch non-integer numbers.

Reply:
![put-not-integer](https://user-images.githubusercontent.com/25161542/163084681-03c5e0b0-3d94-4e5a-a18b-86f22ca3ee4e.png)
![delete-not-integer](https://user-images.githubusercontent.com/25161542/163084689-6e3e9f7e-226d-4e93-b967-b38a22b48b7c.png)